docs: Adds developing readme

nfiniteset opened the pull request:
- Instructions for first time setup
- Notes on architecture and deployment

nfiniteset commented:
Why skip the clean step? Is it because it runs faster?

maxlapides commented:
Yep, you shouldn't need to clean on every build. The Gatsby guides suggest using `gatsby develop`. Not a big deal though.

nfiniteset commented:
Ok I changed the develop script not to run clean. I changed the script in `package.json` rather than changing the readme so people don't have to install Gatsby globally to do the recommended thing.

maxlapides commented:
Okay makes sense, thanks!
